src/runtime/vm/vm.cc

```diff
@@ -118,6 +118,74 @@ Instruction::Instruction(const Instruction& instr) {
   }
 }
 
+Instruction& Instruction::operator=(const Instruction& instr) {
```

Comment:
This will cause leak if `this` originally points to some dynamically allocated memory. Maybe in each case, check  for dynamic memory and release, then call `Duplicate`?

For example:
````cpp
case Opcode::AllocDatatype:
      this->constructor_tag = instr.constructor_tag;
      this->num_fields = instr.num_fields;
      if (this->datatype_fields != nullptr) delete this->datatype_fields;
      this->datatype_fields = Duplicate<RegName>(instr.datatype_fields, instr.num_fields);
      return *this;
````

Reply:
Yeah can do that, we need this for vector initialization